app/src/main/java/org/fossasia/openevent/general/sessions/SessionWithEvent.kt

```diff
@@ -0,0 +1,27 @@
+package org.fossasia.openevent.general.sessions
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import org.fossasia.openevent.general.event.Event
+
+@Entity(
+    primaryKeys = ["event_id", "session_id"],
+    indices = [
+        Index(value = ["event_id"]),
+        Index(value = ["session_id"])
+    ],
+    foreignKeys = [
+        ForeignKey(entity = Event::class,
+            parentColumns = ["id"],
+            childColumns = ["event_id"]),
+        ForeignKey(entity = Session::class,
+            parentColumns = ["id"],
+            childColumns = ["session_id"])
+    ]
+)
+data class SessionWithEvent(
```

Comment:
Why this class no same session can be in two different events  as it is 1:n relationship and not m:n relationship there's no point having this table previously i created this table for speaker because speakers can exist this way but sessions can never exist in this way

Reply:
Thanks for the review. My initial intention was to cache the sessions so that users don't need to fetch sessions multiple time, not to blindly copy speaker section. But I have rethought and I think it might be better to cache them inside the View Model. Let's me know if you have a better suggestion

Reply:
the point is about caching them the point is of the unnecessary table of SessionWithEvent sessions are always unique can't be mapped to two different events

Reply:
Yes I get your point, I'm fixing it

Reply:
And just for reference you don't have to write this query for this table you can simply use relation annotations provided by room and it does all the work by itself i use this query because it's just makes more sense